src/main/daml/Daml/Finance/Instrument/Swap/Fpml/Util.daml

```diff
@@ -137,35 +139,55 @@ getStubRate sc initialStub p cal convention = do
     Some stubRate -> if isSome s.floatingRate then error "both stubRate and floatingRate provided" else Const <$> s.stubRate
     None -> if isSome s.floatingRate then getStubRateFloating (fromSome s.floatingRate) p cal convention else None
 
+-- | Align the payment schedule with the calculation schedule.
+alignPaymentSchedule : [SchedulePeriod] -> [SchedulePeriod] -> Update [SchedulePeriod]
+alignPaymentSchedule calculationSchedule paymentSchedule = do
+  assertMsg "The payment schedule has fewer periods than the calculation schedule" $ length paymentSchedule <= length calculationSchedule
+  let
+    removeStubElements acc p =
+      acc ++ if isSome p.stubType then [] else [p]
+    calculationScheduleWithoutStubs = foldl removeStubElements [] calculationSchedule
+    paymentScheduleWithoutStubs = foldl removeStubElements [] paymentSchedule
+    nCalculationPeriodsPerPaymentPeriod = length calculationScheduleWithoutStubs / length paymentScheduleWithoutStubs
+    paymentScheduleAlignedWithoutStubs = foldl (\acc p -> acc ++ replicate nCalculationPeriodsPerPaymentPeriod p) [] paymentScheduleWithoutStubs
```

Comment:
Here you can use a [`foldMap`](https://docs.daml.com/daml/stdlib/DA-Foldable.html) :) 

```suggestion
    paymentScheduleAlignedWithoutStubs = foldMap (replicate nCalculationPeriodsPerPaymentPeriod) paymentScheduleWithoutStubs
```